### PR TITLE
fix issue 26

### PR DESCRIPTION
**Pull request by koudle**
fix issue #26

### CODE REVIEW

**Comment by kevmoo** [comment]
@mit-mit – would you triage?

**Comment by mit-mit** [comment]
cc @jonasfj

**Review by jonasfj — APPROVED** [review]
Please rebase on your previous PR :)

File: example/src/examples/http_proxy_repository.dart [hunk 1]
```diff
@@ -27,15 +27,19 @@ class HttpProxyRepository extends PackageRepository {
...
+      if (versions != null) {
+        for (var item in versions) {
+          var pubspec = item['pubspec'];
+          var pubspecString = convert.json.encode(pubspec);
+          yield new PackageVersion(pubspec['name'] as String,
+              pubspec['version'] as String, pubspecString);
+        }
       }
+    } else {
+      yield null;
```
Comment (jonasfj) [inline example/src/examples/http_proxy_repository.dart:42]: I don't think we need to `yield null`, just do something like:
```
if (response.statusCode != 200) {
  return;
}
```

I think that'll be fine... then we don't need to filter afterwards...

**Review by jonasfj — CHANGES_REQUESTED** [review]

File: example/src/examples/http_proxy_repository.dart (hunk 1 above)
Comment (kevmoo) [reply example/src/examples/http_proxy_repository.dart:42]: Agreed!

**Comment by mit-mit** [comment]
@koudle thanks for contributing. Are you interested in completing this PR, or shall we close it?

**Comment by koudle** [comment]
@mit-mit yes,I am interested to complete this PR, but there are some conflicts,I'll close it and create a new PR.

**Pull request closed without merging**